datenguide_python/query_builder.py

```diff
@@ -20,32 +25,61 @@ class Field(NamedTuple):
         If the filter is not set, then only the summed result is returned.)
     """
 
-    name: str
-    subfields: list
-    args: Optional[Dict[str, Union[str, List[str]]]] = None
+    def __init__(
+        self,
+        name: str,
+        fields: List[Union[str, object]] = [],
+        args: Optional[Dict[str, Union[str, List[str]]]] = None,
+    ):
+
+        self.name = name
+        self.fields = fields
+        self.args = args
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def add_field(self, field: Union[str, "Field"]) -> object:
+        if isinstance(field, str):
+            added_field = Field(name=field)
+        else:
+            added_field = field
+
+        if self.fields:
+            self.fields.append(added_field)
+        else:
+            self.fields = [added_field]
+        return added_field
+
+    def add_args(self, args: dict):
+        if self.args:
+            self.args.update(args)
+        else:
+            self.args = args
+
+    def get_info(self) -> Optional[TypeMetaData]:
+        return QueryExecutioner().get_type_info(self.name)
```

Comment:
This will not work in general. The return type of a field is not the same as the name of the field. This is only the case for the statistics themselves. Regions for instance have the field "region" and the return type "Region". For "allRegions" the return type is "RegionResults". The spike contained a private attribute for this, which can be automatically added with the use of meta data when a new field is added.

Reply:
in my construct "region" and "allRegions" are always Queries and never Fields. Maybe we can discuss this on Wednesday, how to best construct the different classes?

Reply:
To me this issue is more related to representing graphQL in general, where I think we should not rely on some properties of the dathenguide API that are only true for some (admittedly large) portion of the fields/types. It is not even clear to what extend these are fixed conventions or just chance. But the current field implementation would not be able to get infos for instance for field "source" with type "Source" for all atomic fields like "id", "name" etc. with types "Int" "String" etc. "regions" (from RegionResults) also returns "Region" objects and would not work. I also think that it would be nice if get_info would work on the top level, i.e. for allRegions and for region, particularly because these are starting points for query building.
The "field name" = "return type name" is especially not true for building meta queries which I  would consider nice if we could build these within our own framework, although it doesn't affect the end user whether this is possible or not.
I feel like the increase internal complexity of the field class by means of including the ._returntype is justified by having a get_info method that works in general. Without the ._returntype attribute the Field method appears slightly simpler, but get_info would have to deal with all the edge cases that fall out of the "field name" = "return type name" scenario. I believe that will be less systematic and harder to maintain.